pkg/querier/queryrange/roundtrip.go

```diff
@@ -1039,7 +1059,79 @@ func NewInstantMetricTripperware(
 	}), nil
 }
 
-func NewVolumeTripperware(cfg Config, log log.Logger, limits Limits, schema config.SchemaConfig, merger base.Merger, iqo util.IngesterQueryOptions, c cache.Cache, cacheGenNumLoader base.CacheGenNumberLoader, retentionEnabled bool, metrics *Metrics, metricsNamespace string) (base.Middleware, error) {
+func NewVolumeTripperware(
+	cfg Config,
+	log log.Logger,
+	limits Limits,
+	schema config.SchemaConfig,
+	merger base.Merger,
+	iqo util.IngesterQueryOptions,
+	c cache.Cache,
+	cacheGenNumLoader base.CacheGenNumberLoader,
+	retentionEnabled bool,
+	metrics *Metrics,
+	metricsNamespace string,
+	logTripperware base.Middleware,
+) (base.Middleware, error) {
+	indexTw, err := indexVolumeQueryTripperware(
+		c,
+		cacheGenNumLoader,
+		cfg,
+		iqo,
+		limits,
+		log,
+		merger,
+		metrics,
+		metricsNamespace,
+		retentionEnabled,
+		schema,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	indexTw = volumeFeatureFlagRoundTripper(
+		volumeRangeTripperware(indexTw),
+		limits,
+	)
+
+	return base.MiddlewareFunc(func(next base.Handler) base.Handler {
+		logHandler := logTripperware.Wrap(next)
+
+		return base.HandlerFunc(func(ctx context.Context, req base.Request) (base.Response, error) {
+			r, ok := req.(*logproto.VolumeRequest)
+			if !ok {
+				return nil, httpgrpc.Errorf(
+					http.StatusBadRequest,
+					"invalid request type, expected *logproto.VolumeRequest",
+				)
+			}
+
+			// idxVolumeHandler gets volume from metadata in the index
+			idxVolumeHandler := indexTw.Wrap(next)
+			if !r.AggregatedMetrics || !cfg.AggregatedMetrics {
+				return idxVolumeHandler.Do(ctx, req)
```

Comment:
is it intended to silently not use aggregated metrics when the user explicitly sets "aggregatedMetrics=true" in the request, but does not have patterns enabled?
or should we throw a bad request instead?